search/query/regexp.go

```diff
@@ -76,13 +78,13 @@ func (q *RegexpQuery) Validate() error {
 
 func (q *RegexpQuery) compile() error {
 	if q.compiled == nil {
-		// require that pattern be anchored to start and end of term
+		// require that pattern NOT be anchored to start and end of term
 		actualRegexp := q.Regexp
-		if !strings.HasPrefix(actualRegexp, "^") {
-			actualRegexp = "^" + actualRegexp
+		if strings.HasPrefix(actualRegexp, "^") {
+			actualRegexp = actualRegexp[1:] // remove leading ^
 		}
-		if !strings.HasSuffix(actualRegexp, "$") {
-			actualRegexp = actualRegexp + "$"
+		if strings.HasSuffix(actualRegexp, "$") {
+			actualRegexp = actualRegexp[:len(actualRegexp)-1] // remove trailing $
 		}
```

Comment:
One thought is perhaps the user is looking for '$' (many folks are looking for U.S. dollars?), so wondering if there's some backslash edge case around this?  This probably hardly ever occurs, if at all.